types/README.md

```diff
@@ -0,0 +1,63 @@
+# Container Type Definition Format
+
+This document describes the format of the container definition files contained in this directory.
+
+### info
+- `type_name`
+
+  The fully-qualified name of the container type. This is used to match against
+  the names of types contained in an executable's debug information.
+
+- `ctype`
+
+  A reference to the enum `ContainerTypeEnum` in OI's source code.
+
+- `header`
+
+  The name of the C++ header file in which this container is defined.
+
+- `stub_template_params`
+
+  The indexes of template parameters which do not represent types stored within
+  this container. These parameters will not be recursed into and measured by OI.
+
+- `underlying_container_index`
+
+  Only used for container adapters. Points OI to the template parameter
+  representing the underlying container to be measured.
+
+### codegen
+- `decl`
+
+  C++ code for the declaration of a `getSizeType` function for this container.
+
+- `func`
+
+  C++ code for the definition of a `getSizeType` function for this container.
+
+
+## Changes introduced with TypeGraph
+- `typeName` and `matcher` fields have been merged into the single field `type_name`.
```

Comment:
I don't agree with this change. The idea of `matcher` is to be able to combine `types/string_type.toml` and `types/cxx11_string_type.toml` which are identical excluding the type name, reducing duplication. It hasn't been done yet, but it could get rid of two duplicate type files if used fully. Was there another motivation for removing it?

Reply:
My feeling is that `matcher` is too low level to require users to define themselves. Each regex engine has a slightly different syntax, which I think should be an implementation detail hidden from users.

Regex has some potential pitfalls for users too, if they're not careful (and not everyone's a regex expert), e.g. some mistakes I can see happening:
`matcher = "ns::vector<"` matches "other_ns::vector"
`matcher = "^ns::vector\b"` matches "ns::vector::value_type"

For all the types we currently have, it's possible to automatically generate a regex from the type name, which (I hope) will work correctly.

Deduplicating TOML files is a good goal, but again I think there could be some struggles for users to get the right regex syntax. How about making type_name into a list and automatically generating the regex from the multiple names? e.g.:
`type_names = ["std::basic_string", "std::__cxx11::basic_string"]`